packages/@aws-cdk/aws-ecs/lib/firlens-log-router.ts

```diff
@@ -0,0 +1,204 @@
+  /**
+   * fluentbit
+   */
+  FLUENTBIT = 'fluentbit',
+
+  /**
+   * fluentd
+   */
+  FLUENTD = 'fluentd',
+}
+
+/**
+ * Firelens configuration file type, s3 or file path.
+ * https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_firelens.html#firelens-taskdef-customconfig
+ */
+export enum FirelensConfigFileType {
+  /**
+   * s3
+   */
+  S3 = 's3',
+
+  /**
+   * fluentd
+   */
+  FILE = 'file',
+}
+
+/**
+ * The options for firelens log router
+ * https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_firelens.html#firelens-taskdef-customconfig
+ */
+export interface FirelensOptions {
+  /**
+   * By default, Amazon ECS adds additional fields in your log entries that help identify the source of the logs.
+   * You can disable this action by setting enable-ecs-log-metadata to false.
+   * @default - true
+   */
+  readonly enableECSLogMetadata?: boolean;
+
+  /**
+   * Custom configuration file, s3 or file
+   * @default - determined by checking configFileValue with S3 ARN.
+   */
+  readonly configFileType?: FirelensConfigFileType;
+
+  /**
+   * Custom configuration file, S3 ARN or a file path
+   */
+  readonly configFileValue: string;
+}
+
+/**
+ * Firelens Configuration
+ * https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_firelens.html#firelens-taskdef
+ */
+export interface FirelensConfig {
+
+  /**
+   * The log router to use
+   * @default - fluentbit
+   */
+  readonly type: FirelensLogRouterType;
+
+  /**
+   * Firelens options
+   * @default - no additional options
+   */
+  readonly options?: FirelensOptions;
+}
+
+/**
+ * The properties in a firelens log router.
+ */
+export interface FirelensLogRouterProps extends ContainerDefinitionProps {
+  /**
+   * Firelens configuration
+   */
+  readonly firelensConfig: FirelensConfig;
+}
+
+/**
+ * The options for creating a firelens log router.
+ */
+export interface FirelensLogRouterDefinitionOptions extends ContainerDefinitionOptions {
+  /**
+   * Firelens configuration
+   */
+  readonly firelensConfig: FirelensConfig;
+}
+
+/**
+ * Render to CfnTaskDefinition.FirelensConfigurationProperty from FirelensConfig
+ */
+function renderFirelensConfig(firelensConfig: FirelensConfig): CfnTaskDefinition.FirelensConfigurationProperty {
+  if (!firelensConfig.options) {
+    return { type: firelensConfig.type };
+  } else {
+    const enableECSLogMetadata = firelensConfig.options.enableECSLogMetadata || firelensConfig.options.enableECSLogMetadata === undefined ? 'true' : 'false';
+    const configFileType = firelensConfig.options.configFileType || /arn:aws[a-zA-Z-]*:s3:::.+/.test(firelensConfig.options.configFileValue)
+      ? FirelensConfigFileType.S3 : FirelensConfigFileType.FILE;
+    return {
+      type: firelensConfig.type,
+      options: {
+        'enable-ecs-log-metadata': enableECSLogMetadata,
+        'config-file-type': configFileType,
+        'config-file-value': firelensConfig.options.configFileValue
```

Comment:
it's different from the parameters in CFN, here configFileValue is mandatory, configFileType is optional since it could be detected by configFileValue(s3 ARN or file path), this is recommended by @SoManyHs in #5457 (comment)